samples/next-with-stripe/components/ElementsForm.tsx

```diff
@@ -30,79 +63,56 @@ const ElementsForm: FC<{
-            <h2>Error 😭</h2>
-            <p className="error-message">{errorMessage}</p>
-          </>
-        )
-
-      default:
-        return null
-    }
-  }
-
   const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
     setInput({
       ...input,
       [e.currentTarget.name]: e.currentTarget.value
     })
   }
 
-  // eslint-disable-next-line @typescript-eslint/no-misused-promises
-  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (e) => {
-    e.preventDefault()
-    // Abort if form isn't valid
-    if (!e.currentTarget.reportValidity()) return
-    if (elements == null) return
-    setPayment({ status: 'processing' })
-
-    // Create a PaymentIntent with the specified amount.
-    const response = await fetchPostJSON('/api/payment_intents', {
-      amount: input.customDonation,
-      payment_intent_id: paymentIntent?.id
-    })
-    setPayment(response)
-
-    if (response.statusCode === 500) {
-      setPayment({ status: 'error' })
-      setErrorMessage(response.message)
-      return
-    }
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = (e) => {
+    async function submit () {
+      e.preventDefault()
+      // Abort if form isn't valid
+      if (!e.currentTarget.reportValidity()) return
+      if (elements == null) return
+      setPayment({ status: 'processing' })
+
+      // Create a PaymentIntent with the specified amount.
+      const response = await fetchPostJSON('/api/payment_intents', {
+        amount: input.customDonation,
+        payment_intent_id: paymentIntent?.id
+      })
+      setPayment(response)
+
+      if (response.statusCode === 500) {
+        setPayment({ status: 'error' })
+        setErrorMessage(response.message)
+        return
+      }
 
-    // Use your card Element with other Stripe.js APIs
-    const error = await stripe?.confirmPayment({
-      elements,
-      confirmParams: {
-        return_url: 'http://localhost:3000/donate-with-elements',
-        payment_method_data: {
-          billing_details: {
-            name: input.cardholderName
+      // Use your card Element with other Stripe.js APIs
+      const error = await stripe?.confirmPayment({
+        elements,
+        confirmParams: {
+          return_url: 'http://localhost:4200/donate-with-elements',
+          payment_method_data: {
+            billing_details: {
+              name: input.cardholderName
+            }
           }
         }
-      }
-    })
+      })
 
-    if (error !== undefined) {
-      setPayment({ status: 'error' })
-      setErrorMessage(error?.error.message ?? 'An unknown error occurred')
-    } else if (paymentIntent != null) {
-      setPayment(paymentIntent)
+      if (error !== undefined) {
+        setPayment({ status: 'error' })
+        setErrorMessage(error?.error.message ?? 'An unknown error occurred')
+      } else if (paymentIntent != null) {
+        setPayment(paymentIntent)
+      }
     }
+
+    void submit()
```

Comment:
I've had this pattern bite me couple of times as it ignores possible errors. I would suggest
```suggestion
    submit().catch(() => {})
```
which passes linting as well but is safer